### DISCUSSION TITLE
Validation when changing settings

### DISCUSSION THREAD

**Discussion by nyrik8989** [spatie/laravel-settings#88] [Q&A]
How to validate stored data using standard Laravel validation mechanisms
```
class SettingClientBlockedClear extends Settings
{
    public int $subMonth;
             .....
    public static function rules(): array
    {
        return [
            'subMonth' => 'required|numeric|gt:0',
        ];
    }
}

**Comment by renepardon** [comment]
I've the same question. I use **nuwave/lighthouse** and created a "UpdateSetting" mutation class. It accepts the group, name, locked and payload params but from this alone I can't derive the corresponding settings class which have different locations in the application (because of **nwidard/laravel-modules**.

There is the SettingsContainer which contains all the discovered settings class names but I don't want to initialize them all just to update a single setting.

**Comment by rubenvanassche** [comment]
Since we have a `SavingSettings` event, you could listen to that and add the static rules method as you described above. Since the event has all the properties of the settings class the only thing you need to do is creating a validator based upon the rules within the settings class and validate the properties.

That's how I would solve it! But ideally I would just validate these values in the controller when the request comes in.

**Reply by renepardon** [reply]
Do you also have an idea about my case? Because I have no type hint available for the proper settings class in my case. So based on the settings I might have to derive the settings class somehow to be able to validate the settings. Maybe I need one more coffee to think differently but right now I have no idea on how to solve that problem.

**Reply by rubenvanassche** [reply]
I don't think it's that big of a problem to initialize the settings container like this:

```php
app(SettingsContainer::class);
```

This actually happens when you boot the application and if your settings discovery is cached it won't be a performance hit.
